Twitch Discord Bridge huge refactoring (#5)
ollprogram/TwitchDiscordBridge: README.md

> Hi I'm ollprogram, the author of this project. Thanks for using it. **Feel free to report any bugs or typo. I'll try to fix them. Since english isn't my main language, feel free to correct me if I made any mistake in this documentation.**

Reviewer: ```suggestion
Hi I'm ollprogram, the author of this project. Thanks for using it. **If you spot any bugs, typos, or awkward wording, don’t hesitate to let me know. I’ll do my best to fix them. English isn’t my first language, so any corrections or suggestions are welcome.**
```

Find information about the licence used for this project [here](LICENSE).

Familiarize yourself with the licence before using my project. It gives information about how you can use it.

## Description

With this application you will be able to link your Twitch chat with a Discord text channel in your discord server (guild).

> When someone will send something on a specified channel on Discord, the same message will be sent to a specified Twitch chat, and reversely.

Reviewer: ```suggestion
Whenever someone sends a message in a specific Discord channel, that same message will automatically be forwarded to a designated Twitch chat, and vice versa.
```

>⚠️ This application is designed to process a high volume of messages. However, due to Twitch’s low request rate limit, latency may occur on channels with significant activity. Therefore, using TwitchDiscordBridge on highly active channels is not recommended.

>⚠️ Using TwitchDiscordBridge on channels you do not own or for which you do not have moderator rights is not recommended.

---

> In this project, I'm often referring to a "Bridge". But, what does a "Bridge" represent in this project?

> > A "Bridge" is an object which transfers messages between two destinations. In our case destinations are Discord and Twitch.

Reviewer: A “Bridge” is an object responsible for transferring messages between two destinations : Discord and Twitch, in our case.

> > A "Bridge" is an object which transfers messages between two destinations. In our case destinations are Discord and Twitch.

Reviewer: ```suggestion
> A “Bridge” is an object responsible for transferring messages between two destinations : Discord and Twitch, in our case.
```

## Download

You can download the runnable jar file [here](https://github.com/ollprogram/TwitchDiscordBridge/releases).

## Setup

There are some procedures before launching the program.

### Discord setup

1. You need a [Discord](https://discord.com) account and a Discord server (guild).
2. Register a new bot on the [Discord Developer Portal](https://discord.com/developers/docs/intro) .  
   You should also turn it to private mode.
3. In the installation section, turn on "guild install"
4. In the OAuth2 section, in the OAuth2 URL Generator select the "bot" scope.
   The Discord bot will need some permissions on your server. See the permissions needed in the screenshot below:

   ![Discord Permissions](images/discord_permissions.png)

>    If you want to activate other permissions, you will have to think about consequences. For example, if you decide to activate mentions to all roles for the bot, then every person from Twitch will be able to mention all roles on Discord.

Reviewer: ```suggestion
   If you choose to enable additional permissions, be mindful of their implications. For example, granting the bot permission to mention all roles would allow anyone from Twitch to trigger role-wide mentions on Discord.
```

5. Use the generated URL to add your bot in your discord guild

### Twitch setup

1. Create a new [Twitch](https://www.twitch.tv) account for your Twitch bot. It's recommended to make your bot a moderator on the Twitch channel where you stream.
2. Create a new twitch application on the [twitch developer portal](https://dev.twitch.tv/console) with your bot account

---

### First launch

> When you launch the program for the first time, you need:

Reviewer: Here are the prerequisites for first time launch :

> When you launch the program for the first time, you need:

Reviewer: ```suggestion
Here are the prerequisites for first time launch :
```

     ```
   - Or you can use this token generator online (https://twitchtokengenerator.com/) if you trust it, and if it works for you. 
   - It's also possible to use the Twitch CLI.
3. A Twitch channel name (The channel where you stream).
4. A Discord channel ID. You need to activate first dev mode on Discord `advanced -> developer_mode -> on`. Right-click on the channel where you want your bot to listen to it, and copy the ID.

## Usage

### To launch the program with the console interface

`java -jar TwitchDiscordBridge-1.0-shaded.jar` (cmd, powershell or bash)

### To list all commands

Type `help` in the console.

### All commands on discord

- `/code` to get source code information.
- `/bridge discord_target <channel_id>` change the discord channel target
- `/bridge twitch_target <channel_name>` change the twitch channel target
- `/bridge open` open the bridge.
- `/bridge close` close the bridge.
- `/bridge info` get information about the bridge.

### All commands in the console

- `help` see all commands
- `shutdown` shutdown the app without exceptions (wait all tasks end).
- `shutdown now` shutdown the app with exceptions.

> - `say your message` send a message on both platforms.

Reviewer: ```suggestion
- `say <your message>` send a message on both platforms.
```
